File: app/src/main/java/info/nightscout/androidaps/activities/PreferencesActivity.java
package info.nightscout.androidaps.activities;

import android.os.Bundle;

import androidx.fragment.app.FragmentTransaction;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceScreen;

import info.nightscout.androidaps.R;

public class PreferencesActivity extends NoSplashAppCompatActivity implements
        PreferenceFragmentCompat.OnPreferenceStartScreenCallback {

    int id;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        //AndroidInjection.inject(this);
        setTheme(R.style.AppTheme_NoActionBar);
        super.onCreate(savedInstanceState);
        MyPreferenceFragment myPreferenceFragment = new MyPreferenceFragment();
        id = getIntent().getIntExtra("id", -1);
        Bundle args = new Bundle();
        args.putInt("id", id);
        myPreferenceFragment.setArguments(args);
        getSupportFragmentManager().beginTransaction().replace(android.R.id.content, myPreferenceFragment).commit();
    }


    @Override public boolean onPreferenceStartScreen(PreferenceFragmentCompat caller, PreferenceScreen pref) {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        MyPreferenceFragment fragment = new MyPreferenceFragment();
        Bundle args = new Bundle();
        args.putString(PreferenceFragmentCompat.ARG_PREFERENCE_ROOT, pref.getKey());
        args.putInt("id", id);
        fragment.setArguments(args);
        ft.replace(android.R.id.content, fragment, pref.getKey());
        ft.addToBackStack(pref.getKey());
        ft.commit();
        return true;
    }
}

File: app/src/main/java/info/nightscout/androidaps/activities/PreferencesActivity.kt
package info.nightscout.androidaps.activities

import android.content.Context
import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceScreen
import info.nightscout.androidaps.MainApp
import info.nightscout.androidaps.R
import info.nightscout.androidaps.utils.LocaleHelper

class PreferencesActivity : NoSplashAppCompatActivity(), PreferenceFragmentCompat.OnPreferenceStartScreenCallback {
    var id = 0
    override fun onCreate(savedInstanceState: Bundle?) { //AndroidInjection.inject(this);
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_single_fragment)
        title = MainApp.gs(R.string.nav_preferences)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
        val myPreferenceFragment = MyPreferenceFragment()
        id = intent.getIntExtra("id", -1)
        val args = Bundle()
        args.putInt("id", id)
        myPreferenceFragment.arguments = args
        supportFragmentManager.beginTransaction().replace(R.id.frame_layout, myPreferenceFragment).commit()
    }

    override fun onPreferenceStartScreen(caller: PreferenceFragmentCompat, pref: PreferenceScreen): Boolean {
        val fragment = MyPreferenceFragment()
        val args = Bundle()
        args.putString(PreferenceFragmentCompat.ARG_PREFERENCE_ROOT, pref.key)
        args.putInt("id", id)
        fragment.arguments = args
        supportFragmentManager.beginTransaction()
            .replace(R.id.frame_layout, fragment, pref.key)
            .addToBackStack(pref.key)
            .commit()
        return true
    }

    public override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(LocaleHelper.wrap(newBase))
    }
}